skore/src/skore/dataset_utils.py

```diff
@@ -0,0 +1,315 @@
+                    continue
+                    
+                if stats_data["same_distribution"]:
+                    print(f"- {col}: Similar distributions (p={stats_data['p_value']:.4f})")
+                else:
+                    print(f"- {col}: Different distributions (p={stats_data['p_value']:.4f})")
+                    print(f"  Dataset 1: mean={stats_data['dataset1_mean']:.4f}, std={stats_data['dataset1_std']:.4f}")
+                    print(f"  Dataset 2: mean={stats_data['dataset2_mean']:.4f}, std={stats_data['dataset2_std']:.4f}")
+    
+    return results
+
+def check_data_leakage(
+    report, 
+    new_data,
+    id_column=None, 
+    threshold=0.05,
+    verbose=True
+):
+    """
+    Check if new data overlaps with training data in a report object.
+    
+    This utility helps ensure that when using report_metrics with new data (X_y syntax),
+    the data is actually "new" and doesn't overlap significantly with training data.
+    
+    Parameters
+    ----------
+    report : EstimatorReport or ComparisonReport
+        The report object containing reference to training data
+    new_data : DataFrame or array
+        New data to check for overlap with training data
+    id_column : str, optional
+        Column name to use for identifying unique rows
+    threshold : float, default=0.05
+        Maximum acceptable overlap percentage (0.05 = 5%)
+    verbose : bool, default=True
+        Whether to print comparison summary
+        
+    Returns
+    -------
+    bool
+        True if significant overlap detected (exceeding threshold)
+    dict
+        Detailed leakage report from compare_datasets
+        
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from skore import EstimatorReport
+    >>> from skore.dataset_utils import check_data_leakage
+    >>> 
+    >>> # Create and train model
+    >>> X_train = pd.DataFrame({"feature1": [1, 2, 3], "feature2": [4, 5, 6]})
+    >>> y_train = [0, 0, 1]
+    >>> X_new = pd.DataFrame({"feature1": [2, 3, 4], "feature2": [5, 6, 7]})
+    >>> y_new = [0, 1, 1]
+    >>> 
+    >>> model = RandomForestClassifier()
+    >>> report = EstimatorReport(model, X_train=X_train, y_train=y_train)
+    >>> 
+    >>> # Check for data leakage before evaluating on "new" data
+    >>> has_leakage, leakage_report = check_data_leakage(report, X_new)
+    >>> 
+    >>> if not has_leakage:
+    >>>     report.report_metrics(X_y=(X_new, y_new))
+    >>> else:
+    >>>     print(f"Warning: {leakage_report['overlap_analysis']['overlap_percentage']}% overlap detected")
+    """
+    import warnings
+    
+    # Extract training data from report object
+    if hasattr(report, "X_train") and report.X_train is not None:
+        # EstimatorReport case
+        train_data = report.X_train
+    elif hasattr(report, "reports_") and len(report.reports_) > 0 and hasattr(report.reports_[0], "X_train"):
```

Comment:
I took too long to give you a review; something changed in ComparisonReport: it's now possible to have EstimatorReports with different X_train, as long as the y is the same. Therefore there is no way to know which report the user want to check.  
I think this function, `check_data_leakage` could apply only to EstimatorReport.